### PR TITLE
Update pyopenms recipe to fix zlib issue

### PR DESCRIPTION
**Pull request by timosachsenberg**
Update pyopenms recipe to fix zlib issue

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by bgruening** [comment]
You need to bump the build number.

**Comment by timosachsenberg** [comment]
@jpfeuffer still get an conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("boost-cpp==1.78.0=h75c5d50_1"), MatchSpec("icu[version='>=70.1,<71.0a0']")}

for reference this is the openms recipe: https://github.com/bioconda/bioconda-recipes/pull/41392/files

**Comment by jpfeuffer** [comment]
Not sure if it just takes some time or if this is related to the errors you had during `bioconda-bot fetch artifacts`.

**Comment by jpfeuffer** [comment]
@bgruening It looks like libopenms was never updated. Do you have an idea why?
https://anaconda.org/bioconda/libopenms

**Comment by jpfeuffer** [comment]
[#41824](https://github.com/bioconda/bioconda-recipes/pull/41824) Well we wanted to enable macos anyway again. Let's see if this one uploads.

File: recipes/pyopenms/meta.yaml [hunk 1]
```diff
@@ -10,20 +10,21 @@ source:
 
 build:
   skip: True  # [py2k]
-  number: 2
+  number: 4
```
Comment (jpfeuffer) [inline recipes/pyopenms/meta.yaml:13]:
```suggestion
  number: 3
```

**Comment by timosachsenberg** [comment]
```
critical libmamba Pathname can't be converted from UTF-8 to current locale.
error    libmamba Error opening for reading "/opt/mambaforge/pkgs/python-libarchive-c-5.0-py38h578d9bd_0/info/index.json": No such file or directory
error    libmamba Error when extracting package: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
python-libarchive-c-5.0-py38h578d9bd_0.conda extraction failed
Found incorrect download: python-libarchive-c. Aborting

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/opt/mambaforge/lib/python3.10/site-packages/conda/exceptions.py", line 1124, in __call__
        return func(*args, **kwargs)
      File "/opt/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 941, in exception_converter
        raise e
      File "/opt/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 934, in exception_converter
        exit_code = _wrapped_main(*args, **kwargs)
      File "/opt/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 892, in _wrapped_main
        result = do_call(parsed_args, p)
      File "/opt/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 758, in do_call
        exit_code = create(args, parser)
      File "/opt/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 632, in create
        return install(args, parser, "create")
      File "/opt/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 588, in install
        transaction.fetch_extract_packages()
    RuntimeError: Found incorrect download: python-libarchive-c. Aborting
```

**Comment by timosachsenberg** [comment]
@BiocondaBot please update

**Comment by timosachsenberg** [comment]
@bgruening quick question the error above (see: https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=36338&view=logs&j=eb1ca712-fec5-5d93-deb0-40983fe61f2f&t=8cb71ada-b7e7-54e5-39da-f5f7c855484f&l=744 ) is probably related to an bioconda / bioconda-utils issue? Can we do something about that?

**Comment by timosachsenberg** [comment]
related to https://github.com/conda-forge/python-libarchive-c-feedstock/issues/35

**Comment by bgruening** [comment]
Cool, well done :)

**Pull request merged**